src/ess/dream/io/geant4.py

```diff
@@ -243,18 +244,60 @@ def geant4_load_calibration(filename: CalibrationFilename) -> CalibrationData:
     return CalibrationData(None)
 
 
-def dummy_assemble_detector_data(
+def assemble_detector_data(
     detector: CalibratedBeamline[RunType],
 ) -> DetectorData[RunType]:
-    """Dummy assembly of detector data, detector already contains neutron data."""
-    return DetectorData[RunType](detector)
+    """
+    In the raw data, the tofs extend beyond 71ms, this is thus not an event_time_offset.
+    We convert the detector data to data which resembles NeXus data, with
+    event_time_zero and event_time_offset coordinates.
+
+    Parameters
+    ----------
+    detector:
+        The calibrated detector data.
+    """
+
+    da = detector.copy(deep=False)
+    da.bins.coords["tof"] = da.bins.coords["tof"].to(unit="us")
+
+    period = (1.0 / sc.scalar(14.0, unit="Hz")).to(unit="us")
+    # Bin the data into bins with a 71ms period
+    da = da.bin(tof=sc.arange("tof", 3) * period)
+    # Add a event_time_zero coord for each bin, but not as bin edges,
+    # as all events in the same pulse have the same event_time_zero, hence the `[:2]`
+    da.coords["event_time_zero"] = (
+        sc.scalar(1730450434078980000, unit="ns").to(unit="us") + da.coords["tof"]
```

Comment:
What is this number?

Reply:
It's a random date in the past. The actual value does not matter.
I added a comment.